### PR TITLE
[full-ci] Search all files with ocis backend

### PR DESCRIPTION
**Pull request by JanAckermann**
## Description
Enhancement: Enable search all files for ocis backend

We've enabled the search all files feature for ocis backend:
* Find files in sub directories
* Find files in other places like project spaces

## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPerm2</strong> https://drone.owncloud.com/owncloud/web/25124/65/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingFolderAdvancedPermissionsGroups-shareAdvancePermissionsGroup_feature-L54.png
  

### webUISharingFolderAdvancedPermissionsGroups-shareAdvancePermissionsGroup_feature-L55.png
  

### webUISharingFolderAdvancedPermissionsGroups-shareAdvancePermissionsGroup_feature-L56.png
  

### webUISharingFolderAdvancedPermissionsGroups-shareAdvancePermissionsGroup_feature-L57.png
  

### webUISharingFolderAdvancedPermissionsGroups-shareAdvancePermissionsGroup_feature-L58.png
  

### webUISharingFolderAdvancedPermissionsGroups-shareAdvancePermissionsGroup_feature-L59.png
  


</p></details>
:boom: The acceptance tests pipeline failed. The build has been cancelled.

File: packages/web-app-files/src/components/FilesList/ResourceTable.vue [hunk 1]
```diff
@@ -693,6 +700,21 @@ export default defineComponent({
         resourceType,
         ownerName: resource.owner[0].displayName
       })
+    },
+    getMatchingSpace(storageId) {
+      return this.spaces.find((space) => space.id === storageId)
+    },
+    getDefaultParentFolderName(resource) {
+      if (!this.hasSpaces) {
+        this.$gettext('All files and folders')
```
Comment (JammingBen) [inline packages/web-app-files/src/components/FilesList/ResourceTable.vue:709]:
```suggestion
        return this.$gettext('All files and folders')
```

File: packages/web-app-files/src/components/FilesList/ResourceTable.vue [hunk 2]
```diff
@@ -693,6 +700,21 @@ export default defineComponent({
...
+      }
+      const matchingSpace = this.getMatchingSpace(resource.storageId)
+
+      if (matchingSpace && matchingSpace?.driveType === 'project') {
```
Comment (JammingBen) [inline packages/web-app-files/src/components/FilesList/ResourceTable.vue:713]:
```suggestion
      if (matchingSpace?.driveType === 'project') {
```

File: packages/web-app-files/src/components/Search/Preview.vue [hunk 3]
```diff
@@ -53,9 +55,21 @@ export default {
...
     ...mapGetters(['configuration', 'user', 'getToken']),
+    ...mapState('Files', ['spaces']),
 
+    matchingSpace() {
+      return this.spaces.find((space) => space.id === this.resource.storageId)
+    },
     defaultParentFolderName() {
-      return this.hasSpaces ? this.$gettext('Personal') : this.$gettext('All files and folders')
+      if (!this.hasSpaces) {
+        this.$gettext('All files and folders')
```
Comment (JammingBen) [inline packages/web-app-files/src/components/Search/Preview.vue:65]:
```suggestion
        return this.$gettext('All files and folders')
```

File: packages/web-app-files/src/search/sdk/list.ts [hunk 4]
```diff
@@ -24,7 +24,12 @@ export default class List implements SearchList {
     )
 
     return plainResources.map((plainResource) => {
-      const resource = buildResource(plainResource)
+      let resourceName = decodeURIComponent(plainResource.name)
+      if (resourceName.startsWith('/dav')) {
+        resourceName = resourceName.slice(4)
+      }
+
+      const resource = buildResource({ ...plainResource, ...{ name: resourceName } })
```
Comment (JammingBen) [inline packages/web-app-files/src/search/sdk/list.ts:32]: Could be simplified to this, no?

```suggestion
      const resource = buildResource({ ...plainResource, name: resourceName })
```

File: packages/web-app-files/src/search/sdk/preview.ts [hunk 5]
```diff
@@ -42,7 +42,12 @@ export default class Preview implements SearchPreview {
     return this.cache.set(
       term,
       plainResources.map((plainResource) => {
-        const resource = buildResource(plainResource)
+        let resourceName = decodeURIComponent(plainResource.name)
+        if (resourceName.startsWith('/dav')) {
+          resourceName = resourceName.slice(4)
+        }
+
+        const resource = buildResource({ ...plainResource, ...{ name: resourceName } })
```
Comment (JammingBen) [inline packages/web-app-files/src/search/sdk/preview.ts:50]: Same as above

```suggestion
        const resource = buildResource({ ...plainResource, name: resourceName })
```

**Review by JammingBen — CHANGES_REQUESTED** [review]
Only a few minor things from my side 👍

File: packages/web-app-files/src/components/FilesList/ResourceTable.vue (hunk 2 above)
Comment (JanAckermann) [reply packages/web-app-files/src/components/FilesList/ResourceTable.vue:713]: matching space might be null, don't think optional chaining only will work here

File: packages/web-app-files/src/search/sdk/list.ts (hunk 4 above)
Comment (JanAckermann) [reply packages/web-app-files/src/search/sdk/list.ts:32]: sure ^^

File: packages/web-app-files/src/components/Search/Preview.vue (hunk 3 above)
Comment (JanAckermann) [reply packages/web-app-files/src/components/Search/Preview.vue:65]: :+1:

File: packages/web-app-files/src/components/FilesList/ResourceTable.vue (hunk 1 above)
Comment (JanAckermann) [reply packages/web-app-files/src/components/FilesList/ResourceTable.vue:709]: :+1:

File: packages/web-app-files/src/components/FilesList/ResourceTable.vue (hunk 2 above)
Comment (JammingBen) [reply packages/web-app-files/src/components/FilesList/ResourceTable.vue:713]: Isn't the optional chaining for such cases? I thought if `matchingSpace` is `null`, then you effectively have `if (null === 'project') {...}`.

Comment (JanAckermann) [reply packages/web-app-files/src/components/FilesList/ResourceTable.vue:713]: Let me check but I don’t think it works on null values, just on objects

Comment (JanAckermann) [reply packages/web-app-files/src/components/FilesList/ResourceTable.vue:713]: null?.foo?.bar shouldn’t work, but {}?.foo?.bar should

Comment (JanAckermann) [reply packages/web-app-files/src/components/FilesList/ResourceTable.vue:713]: you was right ;)

**Pull request merged**